src/domain/discounts/discount-form/condition-tables/selectable-table.tsx

```diff
@@ -5,197 +5,183 @@ import {
   ProductTag,
   ProductType,
 } from "@medusajs/medusa"
-import React, { useEffect, useMemo } from "react"
+import { debounce } from "lodash"
+import React, { useEffect } from "react"
 import {
-  ColumnInstance,
+  Column,
+  HeaderGroup,
+  Row,
   usePagination,
   useRowSelect,
   useTable,
 } from "react-table"
 import Spinner from "../../../../components/atoms/spinner"
 import IndeterminateCheckbox from "../../../../components/molecules/indeterminate-checkbox"
-import Table, { TablePagination } from "../../../../components/molecules/table"
-import { PaginationProps } from "../../../../types/shared"
+import Table, {
+  TablePagination,
+  TableProps,
+} from "../../../../components/molecules/table"
+import useQueryFilters from "../../../../hooks/use-query-filters"
 
 type SelectableTableProps = {
-  showSearch?: boolean
-  objectName?: string
+  resourceName?: string
   label?: string
   isLoading?: boolean
-  pagination: PaginationProps
-  totalCount?: number
+  totalCount: number
+  options: Omit<TableProps, "filteringOptions"> & {
+    filters?: Pick<TableProps, "filteringOptions">
+  }
   data?:
     | Product[]
     | ProductType[]
     | ProductCollection[]
     | ProductTag[]
     | CustomerGroup[]
   selectedIds?: string[]
-  columns: Partial<ColumnInstance>[]
-  onPaginationChange: (pagination: PaginationProps) => void
+  columns: Column[]
   onChange: (items: string[]) => void
-  onSearch?: (search: string) => void
-}
+  renderRow: (props: { row: Row }) => React.ReactElement
+  renderHeaderGroup?: (props: {
+    headerGroup: HeaderGroup
+  }) => React.ReactElement
+} & ReturnType<typeof useQueryFilters>
 
 export const SelectableTable: React.FC<SelectableTableProps> = ({
```

Comment:
It would be nice to convert `SelectableTable` to a generic function so we can pass the type to `react-table`

Reply:
very true! pushed a fix, lmk what you think 😄